tests/lib/eslint/flat-eslint.js

```diff
@@ -5448,3 +5449,67 @@ describe("FlatESLint", () => {
     });
 
 });
+
+describe("shouldUseFlatConfig", () => {
+    describe("when the env variable `ESLINT_USE_FLAT_CONFIG` is `'true'`", () => {
+        beforeEach(() => {
+            process.env.ESLINT_USE_FLAT_CONFIG = true;
+        });
```

Comment:
I think this test would be better with a setup where the is no config file present (e.g., with `process.chdir(os.tmpdir())`).

Reply:
Great suggestion, I refactored the test to now test the entire cross product space of env var values x directory, so we should be covered now